src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -227,6 +228,40 @@ private IntPtr GetParentWindow(CoreUIParent uiParent)
             return msalTokenResponse;
         }
 
+        public async Task<MsalTokenResponse> AcquireTokenByUsernamePasswordAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenByUsernamePasswordParameters acquireTokenByUsernamePasswordParameters)
+        {
+            var cancellationToken = authenticationRequestParameters.RequestContext.UserCancellationToken;
+            MsalTokenResponse msalTokenResponse = null;
+
+            _logger.Verbose("[WamBroker] Acquiring token with Username Password flow.");
+
+            using (var core = new NativeInterop.Core())
+            using (AuthParameters authParams = WamAdapters.GetCommonAuthParameters(authenticationRequestParameters, _wamOptions.MsaPassthrough))
+            {
+                authParams.Properties["MSALRuntime_Username"] = acquireTokenByUsernamePasswordParameters.Username;
+                authParams.Properties["MSALRuntime_Password"] = new System.Net.NetworkCredential(string.Empty, acquireTokenByUsernamePasswordParameters.Password).Password;
+
+                using (NativeInterop.AuthResult result = await core.SignInSilentlyAsync(
+                        authParams,
+                        authenticationRequestParameters.CorrelationId.ToString("D"),
+                        cancellationToken).ConfigureAwait(false))
+                {
+                    if (result.IsSuccess)
+                    {
+                        msalTokenResponse = WamAdapters.ParseRuntimeResponse(result, authenticationRequestParameters, _logger);
+                    }
+                    else
+                    {
+                        WamAdapters.ThrowExceptionFromWamError(result, authenticationRequestParameters, _logger);
+                    }
+                }
+            }
+
+            return msalTokenResponse;
```

Comment:
Am curious about the following: 

1. is there an RT given back to the app if the flow works?
2. does it work with POP?

Reply:
Implemented the POP and it works with username and password and returns the pop token